controllers/storagecluster/cephcluster.go

```diff
@@ -1589,24 +1589,89 @@ func generateCephReplicatedSpec(initData *ocsv1.StorageCluster, poolType string)
 	return crs
 }
 
-func getCephClusterCephConfig(sc *ocsv1.StorageCluster) map[string]map[string]string {
+func getCephClusterCephConfig(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) map[string]map[string]string {
+	// Default Ceph configuration settings
 	cephConfig := map[string]map[string]string{
 		"global": {
-			"mon_max_pg_per_osd":    "1000",
-			"mon_target_pg_per_osd": "200",
+			"mon_max_pg_per_osd":                 "1000",
+			"mon_target_pg_per_osd":              "200",
+			"mon_pg_warn_max_object_skew":        "0",
+			"mon_data_avail_warn":                "15",
+			"mon_osd_full_ratio":                 ".85",
+			"mon_osd_backfillfull_ratio":         ".80",
+			"mon_osd_nearfull_ratio":             ".75",
+			"bdev_flock_retry":                   "20",
+			"bluestore_prefer_deferred_size_hdd": "0",
+			"bluestore_slow_ops_warn_lifetime":   "0",
+		},
+		"osd": {
+			"osd_memory_target_cgroup_limit_ratio": "0.8",
 		},
 	}
 
-	// Set the mon_target_pg_per_osd based on the resource profile
+	// Set mon_target_pg_per_osd based on the resource profile
 	switch sc.Spec.ResourceProfile {
 	case "lean":
 		cephConfig["global"]["mon_target_pg_per_osd"] = "100"
 	case "performance":
 		cephConfig["global"]["mon_target_pg_per_osd"] = "400"
 	}
 
+	// Set mon_warn_on_pool_no_redundancy to false only if non-resilient pools are enabled
+	if sc.Spec.ManagedResources.CephNonResilientPools.Enable {
+		cephConfig["global"]["mon_warn_on_pool_no_redundancy"] = "false"
+	}
+
+	// Configure public network if the cluster is dualstack, but not multus
+	if sc.Spec.Network != nil && sc.Spec.Network.Provider == "" && sc.Spec.Network.DualStack {
+		networkConfig := &configv1.Network{}
+		err := r.Client.Get(context.TODO(), types.NamespacedName{Name: "cluster", Namespace: ""}, networkConfig)
+		if err == nil {
+			cidrNameArray := []string{}
+			for _, cidr := range networkConfig.Status.ClusterNetwork {
+				cidrNameArray = append(cidrNameArray, cidr.CIDR)
+			}
+			if len(cidrNameArray) != 0 {
+				cidrName := strings.Join(cidrNameArray, ",")
+				cephConfig["global"]["public_network"] = cidrName
+			}
+
+		}
+		r.Log.Error(err, "could not get network config details", "CephCluster", klog.KRef(sc.Namespace, sc.Name))
+	}
+
+	// Add RBD mirror debug logging configs if RBD mirror CR is present
+	partialList := &metav1.PartialObjectMetadataList{}
+	partialList.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "ceph.rook.io",
+		Version: "v1",
+		Kind:    "CephRBDMirrorList",
+	})
+	err := r.Client.List(context.TODO(), partialList,
+		client.InNamespace(sc.Namespace),
+		client.Limit(1),
+	)
```

Comment:
Have used partialObjectMetadata to list as I just need to be aware of the CRs presence and I don't need it's contents. So have tried to make the call as minimal as possible.